### ISSUE TITLE
EnforcedShorthandSyntax for keyword argument values

### ISSUE THREAD

**Issue by erickgskovik** [rubocop/rubocop#12149]
## What is now for shorthand syntax for keyword argument values.

Ruby 3.1 allows to invoke a method with `bar(test:)` as shorthand for `bar(test: test)`. Here's an example in context:

 ```
class A
  def self.foo(test)
    bar(test:)
  end

  def self.bar(test:)
    puts test
  end
end

puts A.foo('test') # => test
```

`Style/HashSyntax` has `EnforcedShorthandSyntax` options to verify hash. But the rule doesn't apply for arguments.
This applies to:
```▸ rubocop -V

1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
```

## Expected

I can see an use case that differentiate styles between hash and keyword argument values. Is it okay to contribute a new style for argument values? Or enforce such rules on `Style/HashSyntax`?

**Comment by koic** [comment]
Keyword arguments with `EnforcedShorthandSyntax: always (default)` are detected as shown below:

```console
$ bundle exec rubocop -v
1.63.4
$ echo 'm(kwarg: kwarg)' | bundle exec rubocop --stdin example.rb --only Style/HashSyntax -a
Inspecting 1 file
C

Offenses:

example.rb:1:10: C: [Corrected] Style/HashSyntax: Omit the hash value.
m(kwarg: kwarg)
         ^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
====================
m(kwarg:)
```

If this is not the expected behavior, can you provide examples of the expected code and actual code, along with steps to reproduce, to clarify the issue?

**Comment by erickgskovik** [comment]
Sorry, I may not be clear on my expectation.

```sh
$ bundle exec rubocop --version
1.63.5
```

The expect code would be:

```ruby
def m(kwarg:)
end

kwarg = []
m(kwarg: kwarg)
```

The actual code is:
```ruby
def m(kwarg:)
end

kwarg = []
m(:kwarg => kwarg)
```

Rubocop configuration:

```yaml
Style/HashSyntax:
  Enabled: true
  EnforcedStyle: hash_rockets
  EnforcedShorthandSyntax: never
```

To reproduce this,
1. Start with the expected code and configuration.
2. Run `bundle exec rubocop expected.rb -a`. Rubocop output:
```sh
bundle exec rubocop -a test.rb
Inspecting 1 file
C

Offenses:

test.rb:5:3: C: [Corrected] Style/HashSyntax: Use hash rockets syntax.
m(kwarg: kwarg)
  ^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

My main point is that I would like to have hash rocket for hashes. But when it comes to keyword arguments, I would like to use keyword arguments without shorthand syntax.

**Comment by cllns** [comment]
I agree that these should be separated, mostly because it's an inaccurate name. Kwargs are not hashes in Ruby 3, so the should be separate. It also makes sense to have different styles for each, since they occur in very different situations. Perhaps `Style/KeywordArgumentsSyntax`?

**Comment by erickgskovik** [comment]
I see the differences rubocop is making. `Style/KeywordArgumentsSyntax` looks reasonable to me.

**Comment by cllns** [comment]
Commenting to remove `stale` tag. I still think this should be considered

**Comment by cllns** [comment]
Is this a desirable change? I'm happy to work on on a PR if so.
